File: tools/bench/answers/go_haiku/s3/08_invoice.go
package main

import (
	"fmt"
)

func main() {
	type Item struct {
		name      string
		qty       int
		unitPrice float64
	}
	items := []Item{
		{"widget", 4, 2.5},
		{"gizmo", 2, 7.25},
		{"bolt", 10, 0.1},
	}
	var total float64
	var maxItem Item
	var maxLineTotal float64
	for _, item := range items {
		lineTotal := float64(item.qty) * item.unitPrice
		total += lineTotal
		if lineTotal > maxLineTotal {
			maxLineTotal = lineTotal
			maxItem = item
		}
	}
	fmt.Println(total)
	fmt.Println(maxItem.name)
}

File: tools/bench/answers/python_haiku/s3/08_invoice.py
items = [
    ("widget", 4, 2.5),
    ("gizmo", 2, 7.25),
    ("bolt", 10, 0.1)
]
grand_total = 0
max_item = None
max_line_total = 0
for name, qty, price in items:
    line_total = qty * price
    grand_total += line_total
    if line_total > max_line_total:
        max_line_total = line_total
        max_item = name
print(grand_total)
print(max_item)
